api/proxy/proxy.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package proxy;
+
+service Proxy {
+    // Upload will store a backup file in cloud storage
+    rpc Upload (UploadRequest) returns (UploadReply) {}
+    // Download will retrieve a backup file from cloud storage
+    rpc Download (DownloadRequest) returns (DownloadReply) {}
+}
+
+message UploadRequest {
+    // The cluster identifier is a string that identifies this etcd cluster. It should be unique and stable between
+    // backups and restores.
+    string cluster_identifier = 1;
+
+    // This is the time when the backup was actually made. This may well be different from the time it was uploaded.
+    google.protobuf.Timestamp backup_made_at = 2;
+
+    // This is the binary contents of the backup itself.
+    bytes backup = 3;
+}
+
+message UploadReply {
+    // This is the full URL of the backup in remote storage. The structure of this URL is entirely determined by the
+    // proxy and may change from backup to backup, users should not attempt to parse data out of the URL.
+    string backup_url = 1;
+}
+
+message DownloadRequest {
+    // This is the URL of the backup to download.
+    string backup_url = 1;
+}
+
+message DownloadReply {
+    // This is the binary contents of the backup itself.
```

Comment:
It may be worth adding some details of the format here and in the `UploadRequest` object too. Is this a `.tar.gz` of the backup content? Is any metadata required in that?

Reply:
It's just the binary output of the backup. So what would be in a `snapshot.db` file if you did `ETCDCTL_API=3 etcdctl snapshot save --endpoints my-cluster-url snapshot.db`.